spec/rubocop/cop/lint/side_effects_in_notes_spec.rb

```diff
@@ -0,0 +1,166 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'rubocop/cop/lint/side_effects_in_notes'
+
+RSpec.describe RuboCop::Cop::Lint::SideEffectsInNotes do
+  subject(:cop) { described_class.new(config) }
+  let(:config) { RuboCop::Config.new }
+
+  it 'requires Notes to be present when keys are present' do
+    expect_offense(<<~RUBY)
+      class DummyModule
+        def initialize(info = {})
+          super(
+            update_info(
+              info,
+              'Name' => 'Simple module name',
+              'Description' => 'Lorem ipsum dolor sit amet',
+              'Author' => [ 'example1', 'example2' ],
+              'License' => MSF_LICENSE,
+              'Platform' => 'win',
+              'Arch' => ARCH_X86,
+              ^^^^^^ Module is missing the Notes section [...]
+            )
+          )
+        end
+      end
+    RUBY
+    expect_no_corrections
+  end
+
+  it 'requires Notes to be present when no keys are present' do
+    expect_offense(<<~RUBY)
+      class DummyModule
+        def initialize(info = {})
+          super(
+            update_info(
+              info,
+              {}
+              ^^ Module is missing the Notes section [...]
```

Comment:
We should probably link to the wiki page for more details to help the average user understand what this means
https://github.com/rapid7/metasploit-framework/wiki/Definition-of-Module-Reliability,-Side-Effects,-and-Stability

That aside, the wiki page also needs a bit of direction in it too 😄

Reply:
An example on that page would be nice, I always have to look up old modules.
Prob a good idea to update the example modules as well

Reply:
> you can also abbreviate offense messages with `[…]`
> https://www.rubydoc.info/gems/rubocop/RuboCop/RSpec/ExpectOffense

Reply:
> Note that I did reorder module traits to stability, reliability, and then side effects due to the following thought process:
>
> 1. Will I crash anything if I run this?
> 2. Can I expect to get a shell?
> 3. What might need to be cleaned up?

#11785